fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingScrubDangling.java

```diff
@@ -191,8 +191,11 @@ private CompletableFuture<FDBStoredRecord<Message>> deleteIndexIfDangling(FDBRec
         final IndexingCommon.IndexContext indexContext = common.getIndexContext();
         final IndexEntry indexEntry = indexEntryResult.get();
         if (indexContext.isSynthetic) {
-            return store.loadSyntheticRecord(indexEntry.getPrimaryKey()).thenApply(syntheticRecord -> {
-                if (syntheticRecord.getConstituents().isEmpty()) {
+            return store.loadSyntheticRecord(indexEntry.getPrimaryKey()).handle((syntheticRecord, ex) -> {
+                // The null hypothesis is that it is not dangling.
+                final boolean isDangling = ((findException(ex, RecordDoesNotExistException.class) != null) ||
+                                                  (syntheticRecord != null && syntheticRecord.getConstituents().isEmpty()));
+                if (isDangling) {
                     // None of the constituents of this synthetic type are present, so it must be dangling
```

Comment:
I believe this will identify anything where _any_ of the constituents is not found as dangling, which is different from the comment. It probably is the case that we want that behavior, I think.

It is a bit weird that this uses exception handling as control flow, though. If we wanted to fix that, I think we'd need a new variant of `loadSyntheticRecord` that takes an `IndexOrphanBehavior` argument, or something like that.

Reply:
✅